### ISSUE TITLE
FR: All-in-one search mode

### ISSUE THREAD

**Issue by mnaoumov** [tadashi-aikawa/obsidian-another-quick-switcher#152]
Currently you have to choose between multiple commands such as `File name search` or `Recent search`. But you have to choose it beforehand. It would be much more practical if you just have a single command where you type your query and change the mode from there via buttons or special query syntax

**Comment by tadashi-aikawa** [comment]
Hi, @mnaoumov :)

> It would be much more practical if you just have a single command where you type your query and change the mode from there via buttons or special query syntax

We can already use it by the "Command prefix" settings. In the case of the below example,

![](https://raw.githubusercontent.com/tadashi-aikawa/obsidian-another-quick-switcher/master/demo/search-commands-setting.png)

Input `:e hoge` will search "hoge" as "Recent search" even when the dialog didn't start with "Recent search" command.

https://user-images.githubusercontent.com/9500018/223437652-244ccf97-267e-422c-917b-88f56cba71db.mp4

**Comment by mnaoumov** [comment]
@tadashi-aikawa thanks for the feedback. This is already super useful. But I have a few UX suggestions.

- A single keystroke that switches search modes circularly. Equivalent to [Search Everywhere](https://www.jetbrains.com/help/idea/searching-everywhere.html) feature in JetBrains products (ReSharper, IntelliJ IDEA, WebStorm, Pycharm etc)
- Visible prefix such as `:e` shown, so you can get used to them
- Ability to switch between search modes by clicking in dropdown

**Comment by tadashi-aikawa** [comment]
@mnaoumov 
Thank you for suggestions :)

Right. They would be useful because we may want to switch searches after entering a query. I want more opinions from some users. Then, of course, I'll consider the specification of the feature 👍

**Comment by DivineMK** [comment]
Yes I would really really want this feature as it is the only thing I'm missing from while using Quick Switcher++. This would be a great addition to the plugins.
The preview mode and extensive customization of this plugin is too good to miss out on regardless. Thank you very much for such a great plugin!!

**Comment by DivineMK** [comment]
I want to add that it would be great to also include Grep mode into this all-in-one search mode as currently it doesn't have a command prefix like other searches

**Comment by iiz00** [comment]
ほぼ同じような要望だと思うのでここに書き込ませていただきます。自分も
「Search commandの一覧から選んだ設定に基づいて検索を実行、既にAQSが実行中なら最後のクエリを引き継ぐ」機能/コマンドを希望します（細かく考えると、Search commanのリストから選択して検索するコマンドと、検索結果モーダルにおいてクエリを引き継いでSearch commandをリストから選んで再検索する機能の2つに分かれ、後者だけで充分かもしれません）。
- 検索結果をみて後から検索条件を変更できる
- ホットキーをたくさん割り当てたり覚える必要がない

などのメリットがあるかと思います。
 
- タイトル検索でヒットしなかったら見出しやディスクリプションの検索
- ヒット数が多かったらwebクリップやAIチャットログを貯めてあるフォルダを除外した設定で再検索
みたいな動作をさせたいです。

I believe my request is closely related to the original post, so I’m adding it here rather than opening a new issue.
I would also like to request a feature that allows users to execute a search based on a selected Search Command from the list, carrying over the last query if AQS is already running.

This would offer several benefits:

- It allows users to refine search conditions after seeing the initial results.
- It reduces the need to assign or memorize many hotkeys.

For example:

- If a title-based search returns no results, I could easily switch to a Search Command that targets headings or descriptions.
- If the result count is too high, I could re-run the same query with a command that excludes folders where I store web clips or AI chat logs.

This kind of workflow would make AQS even more flexible and powerful.
